src/main/java/mods/railcraft/common/util/network/PacketCurrentItemNBT.java

```diff
@@ -39,15 +39,22 @@ public void writeData(RailcraftOutputStream data) throws IOException {
     @Override
     public void readData(RailcraftInputStream data) throws IOException {
         try {
-            ItemStack stack = data.readItemStack();
+            ItemStack stack = data.readItemStack(), currentItem;
```

Comment:
Just because it seemed like the correct fix would be to get the right `currentItem` from the start (in the constructor), but I couldn't find a way to do it without the one in the incoming packet, so I opted to keep the diff small for now. If this is the solution that is used, I agree it would be best to pick a new name (and/or make wider changes, since the `getActiveItemStack()` call on the server isn't helping).